pyroomacoustics/libroom_src/room.cpp

```diff
@@ -693,13 +693,18 @@ bool Room<D>::scat_ray(
     {
       // As the ray is shot towards the microphone center,
       // the hop dist can be easily computed
-      float hop_dist = (hit_point - mic_pos).norm();
+      Vectorf<D> hit_point_to_mic = mic_pos - hit_point;
+      float hop_dist = hit_point_to_mic.norm();
       float travel_dist_at_mic = travel_dist + hop_dist;
 
       // compute the scattered energy reaching the microphone
       float m_sq = mic_radius * mic_radius;
       float h_sq = hop_dist * hop_dist;
-      Eigen::VectorXf scat_trans = wall.scatter * transmitted * 2.f * m_sq / h_sq;
+      float p_hit_equal = 1.f - sqrt(1.f - m_sq / h_sq);
+      // cosine angle should be positive, but could be negative if normal is
+      // facing out of room so we take abs
+      float p_lambert = 2 * abs(wall.cosine_angle(hit_point_to_mic));
```

Comment:
@fakufaku I added this Wall method for computing the cosine angle. Whether the normal is facing in or out, I think this should work since we can either multiply the normal by -1 to flip its direction if it's facing outward or take the absolute value of the cosine angle.

Reply:
Great! But could you check wether the normal is ourward pointing ? It is better that the result of `wall.cosine_angle` is well defined. In this particular case, the `abs` is fine though.

Reply:
Should the `abs` be moved into `wall.cosine_angle`? I suppose we would mainly be concerned with angles inside the room, and I could make a comment that the method computes the cosine angle assuming the normal is pointing towards the same half-space as the provided vector, therefore returning values within [0, 1].

Because from just a wall, I don't think you can tell whether the normal is pointing in or out? as you need info about the other walls or assume that the provided vector is pointing into the room.